packages/ruby/spec/http_response_spec.rb

```diff
@@ -0,0 +1,67 @@
+require "http_response"
+
+RSpec.describe HttpResponse do
+  describe "#body" do
+    it "returns an empty string when the body is nil" do
+      response = HttpResponse.from_parts(200, {}, nil)
+
+      expect(response.body).to eq ""
+    end
+
+    it "returns an empty string when the body is an empty array" do
+      response = HttpResponse.from_parts(200, {}, [])
+
+      expect(response.body).to eq ""
+    end
+
+    it "returns an empty string when the body is an array of string" do
+      response = HttpResponse.from_parts(200, {}, ["body1", "body2"])
+
+      expect(response.body).to eq "body1body2"
+    end
+
+    it "concatenates the body when it is StringIO" do
+      body = "BODY"
+      response = HttpResponse.from_parts(200, {}, StringIO.new(body))
+
+      expect(response.body).to eq body
+    end
+
+    it "can be read safely multiple times" do
+      body = "BODY"
+      response = HttpResponse.from_parts(200, {}, StringIO.new(body))
+
+      expect(response.body).to eq body
+      expect(response.body).to eq body
+    end
+  end
+
+  describe "#content_length" do
+    it "returns 0 for a response with an empty body status" do
+      response = HttpResponse.from_parts(
+        204,
+        {"Content-Length" => 53},
+        nil
```

Comment:
Can we put an actual body here so the test proves that it's the 204 status rather than the `nil` causing the 0?